packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp

```diff
@@ -155,21 +155,29 @@ namespace { // (anonymous)
          * the number of replaced diagonal entries to match the
          * local number of rows.
          */
-        TEST_EQUALITY(numReplacedDiagEntries, matrix->getNodeNumRows());
+	const LO lclNumRows = static_cast<LO> (matrix->getNodeNumRows ());
+        TEST_EQUALITY(numReplacedDiagEntries, lclNumRows);
 
         /* Test for successful replacement
          *
          * 1. Extract diagonal copy
          * 2. Test if diagonal element matches rank ID the we intended to set
          */
 
-        RCP<vec_type> diagCopy = rcp(new vec_type(matrix->getRowMap()));
-        matrix->getLocalDiagCopy(*diagCopy);
-
-        Teuchos::ArrayRCP<const Scalar> diagCopyData = diagCopy->getData();
-
-        for (size_t i = 0; i < diagCopyData.size(); ++i)
-          TEST_EQUALITY_CONST(diagCopyData[i], rankAsScalar);
+        vec_type diagCopy (matrix->getRowMap ());
```

Comment:
deep copy of the row map?

Reply:
@bartlettroscoe Nope -- `getRowMap()` returns `RCP<const Map>`.

Reply:
Right.  I forgot Tpetra moved to shallow copy semantics.